webhook/resources/volume/mutator.go

```diff
@@ -180,6 +180,16 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/backupCompressionMethod", "value": "%s"}`, defaultCompressionMethod))
 	}
 
+	if volume.Spec.SnapshotMaxCount == 0 {
+		snapshotMaxCount, err := v.getSnapshotMaxCount()
+		if err != nil {
+			err = errors.Wrap(err, "BUG: cannot get valid number for setting snapshot max count")
+			return nil, werror.NewInvalidError(err.Error(), "")
+		}
```

Comment:
We should do the upgrade path as well to make the existing resources to update during the upgrade. Go check upgrade package.

Reply:
Yes, I modify `SnapshotMaxCount` to `250` in upgrade.go file, so we don't hit this one.

Reply:
I remove the change currently, because we can't get latest setting in `Upgrade` function when we just started a new longhorn-manager. We call `Upgrade` function before we create default setting (`UpdateCustomizedSettings `). I think we can just leave mutator to handle it.
Ref: https://github.com/longhorn/longhorn-manager/blob/420b7798b3c5b2d747e391aa6065b3c2695d07fe/app/daemon.go#L162-L185

Reply:
For backward compatibility, we need to make sure if no snapshot max count is set for any existing volumes, the behavior should be the same, with the default limit (internal, it should be 250). 

Then, your original code in the upgrade should be OK. Right?

Reply:
Yeah, let me set a global variable to make sure it's consistent with the default setting. Thank you.

Reply:
Updated it. Thanks.